services/users.service.ts

```diff
@@ -48,7 +45,7 @@ export default class UserService {
             password: hashedPassword,
             firstName: user.firstName,
             lastName: user.lastName,
-            roles: this.roles[2] // default role is (User)
+            roles: userRoles.user
```

Comment:
youre assigning a single value here instead of an array, but you have defined in the schema that roles field is an array